public/preload.js

```diff
@@ -16,6 +16,26 @@ Sentry.init({
 })
 
 process.once('loaded', () => {
+  const oldAccessDataRaw = localStorage.getItem('wallet:accessData');
+  if (oldAccessDataRaw) {
+    // When migrating old wallets initialized with hardware wallets we need to clean the localStorage
+    const accessData = JSON.parse(oldAccessDataRaw);
+    const isHardware = localStorage.getItem('wallet:type') === 'hardware';
+    if (accessData.from_xpub || isHardware) {
+      const uniqueId = localStorage.getItem('app:uniqueId');
+      const backup = localStorage.getItem('wallet:backup');
+      const network = localStorage.getItem('wallet:network');
+      const server = localStorage.getItem('wallet:server');
+      localStorage.clear();
+      localStorage.setItem('localstorage:started', 'true');
+      localStorage.setItem('localstorage:backup', backup);
+      localStorage.setItem('localstorage:network', network);
+      localStorage.setItem('localstorage:server', server);
+      localStorage.setItem('app:uniqueId', uniqueId);
+    }
```

Comment:
I was thinking about moving this to a separate function just to organize the code. Then we can have `migrate_from_vX.Y()` and below we can have a different migration function in the future, so they are not mixed when we need to add more storage migrations.

What do you think?

Reply:
Maybe, but this is not a migration since we are not porting the access data.
We are just clearing the localStorage and keeping some basic flags which could also be removed.